### PR TITLE
Unify the MakeServicesPublicPass

### PR DESCRIPTION
**Pull request by bytehead**
See https://github.com/contao/contao/pull/3743#discussion_r759472200

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]
I think this is the correct way to go. Aliases are exactly here to make sure that old service references still work. So for our compiler pass it shouldn't matter which one is a service and which one is an alias. They just have to end up being public.

File: core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php [hunk 1]
```diff
@@ -22,43 +22,34 @@
...
             }
 
-            $definition = $container->getDefinition($service);
-            $definition->setPublic(true);
-        }
-
-        foreach (self::ALIASES as $alias) {
-            if (!$container->hasAlias($alias)) {
-                continue;
+            if ($container->hasDefinition($id)) {
```
Comment (leofeyer) [inline core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php]: Should this be an `elseif` condition, so we do not look for definitions if the ID is an alias?

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php]: I don't like `elseif`, if the `$id` is an alias then `$container->hasDefinition($id)` will always evaluate to false.

Comment (leofeyer) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php]: Yes, but if we already know that the ID is an alias, the `hasDefinition()` call is unnecessary, isn't it?

File: core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php [hunk 2]
```diff
@@ -22,43 +22,34 @@
...
 
     public function process(ContainerBuilder $container): void
     {
-        foreach (self::SERVICES as $service) {
-            if (!$container->hasDefinition($service)) {
-                continue;
+        foreach (self::IDS as $id) {
+            if ($container->hasAlias($id)) {
+                $alias = $container->getAlias($id);
+                $alias->setPublic(true);
```
Comment (bytehead) [inline core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:46]:
```suggestion
                $alias->setPublic(true);
                
                continue;
```

File: core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php (hunk 1 above)
Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php]: Maybe like https://github.com/contao/contao/pull/3759#discussion_r760963494?

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php]: It's IMO a personal thing. My eyes always hurt a bit if I see `elseif` when not really needed 😅

File: core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:46]: That's even worse IMHO. No hard feelings. 😄

**Review by leofeyer — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**